feat: rework helm chart

okgolove opened the pull request:
resolves #2

* Change repo structure (to use chart-test and chart-release Github Actions, would be glad to implement this)
* Make helm chart more adjustable
* Add useful metadata to Kubernetes workloads
* Use main OliveTin project docker repository
* Fix OliveTin Helm installation doc URL
* Allow creating extra manifests
* Temporary removing helm values schema (I would like to add scheme autogeneration in using Github Actions)

okgolove committed:
feat: rework helm chart

* Change repo structure (to use chart-test and chart-release github actions)
* Make helm chart more adjustable
* Add useful metadata to Kubernetes workloads
* Use main OliveTin project docker repository
* Fix OliveTin Helm installation doc URL
* Allow creating extra manifests

okgolove commented:
Good morning @jamesread! Sorry for tagging, don't have another options to notify you :)

jamesread commented:
Heya, quick reply on my phone - I saw this as soon as I woke up at 7am, and I'm really excited for it :-) I had a quick skim on my phone, but had to do the school runs with kids and day job work - so not had the time to look in detail yet. 

I almost certainly should have time to review in detail this evening! 

You can continue to ping me here, also lots of contact options on my GitHub profile, or https://jread.com, or chat on the OliveTin discord server! :-)

okgolove committed:
fix: use peroprly configmap name in volume

okgolove committed:
feat: do not recreate pod on config change

Now OliveTin supports live reloading, it does not make any sense to recreate the pod

okgolove committed:
feat: bump helm chart version

jamesread commented on charts/olivetin/Chart.yaml:
```diff
@@ -11,9 +11,9 @@ maintainers:
+version: 1.0.0
 
-appVersion: "latest"
+appVersion: "2025.4.8"
```
This would mean the helm chart needs to be updated with every release, right? Isn't this normal? It will create a lot more work.

okgolove replied:
Dependabot / Renovate /  updatecli should help here, I believe. I see your point but as an multiple Kubernetes clusters active maintainer I would prefer receiving more granular and controlled updates. In the same time, setting static `image.tag` value on user side may cause `service <> chart` incompatibility.

jamesread replied:
Ah, if Dependabot can update it I'd be more than happy with that solution - I use dependabot a lot. 

I've not used renovate before but heard good things about it. 

If you have this working in other repos, could you add a GitHub workflow for that, after I merge this PR?

okgolove replied:
Example https://github.com/okgolove/OliveTin-HelmChart/pull/3


I used my own token (usually Renovate Github App is used) and this config (not the best one, just for demo):
```json
{
    "labels": [
        "renovate"
    ],
    "includePaths": [
        "**/charts/olivetin/Chart.yaml"
    ],
    "repositories": [
        {
            "repository": "okgolove/OliveTin-HelmChart",
            "customManagers": [
                {
                    "customType": "regex",
                    "fileMatch": [
                        "(^|/)Chart\\.yaml$"
                    ],
                    "matchStrings": [
                        "\\sappVersion: (?<currentValue>.*)\\s"
                    ],
                    "depNameTemplate": "OliveTin/OliveTin",
                    "datasourceTemplate": "github-releases"
                }
            ]
        }
    ]
}
```

Unfrotunately, Renovate does not support [auto bumping helm chart version using regex manager yet](https://github.com/renovatebot/renovate/pull/34023), but it can be implemented in a Github workflow. Renovate creates a PR -> Workflow autobumps, lets say, minor chart version -> Tests green -> automerge -> helm chart is published by Github workflow (I am working on this one now)

jamesread commented:
Hey @okgolove thanks very much indeed for this PR, you've done a great job here of making it a lot more extensible and probably more standardized from the version I wrote ages ago. 

I just had one comment about the appVersion - I'm conscious about not wanting to update the jelm chart every time I release OliveTin - I think the default should be latest.
